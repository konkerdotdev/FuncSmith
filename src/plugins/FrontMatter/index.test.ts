import * as P from '@konker.dev/effect-ts-prelude';

import * as fixturesFs from '../../test/fixtures/fileset-1';
import * as unit from './index';

describe('plugins', () => {
  describe('FrontMatter', () => {
    it('should return the expected value', async () => {
      const pluginStack = P.pipe(P.Effect.succeed, unit.frontMatter());
      const actual = await P.Effect.runPromise(P.pipe([...fixturesFs.TEST_FILE_SET_1], pluginStack));
      expect(actual).toMatchSnapshot('plugin-front-matter-1');
    });
  });
});
