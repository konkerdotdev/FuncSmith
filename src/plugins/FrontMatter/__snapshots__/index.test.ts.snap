// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins FrontMatter should return the expected value: plugin-front-matter-1 1`] = `
[
  {
    "_id": "e57ec8617d467edfff00943eac189e0ec6eb1875",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      65,
    ],
    "fileBase": "a",
    "fileExt": ".txt",
    "fileName": "a.txt",
    "path": "/tmp/foo/a.txt",
    "relDir": ".",
    "relPath": "a.txt",
  },
  {
    "_id": "39c8584020d82dc00bda6e549a6c55284dd2f5f3",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      66,
    ],
    "fileBase": "b",
    "fileExt": ".txt",
    "fileName": "b.txt",
    "path": "/tmp/foo/b.txt",
    "relDir": ".",
    "relPath": "b.txt",
  },
  {
    "_id": "c7712f15959be6b0c2ad22b5b71d2e367d6ff138",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      98,
      97,
      109,
      44,
      98,
      97,
      122,
      10,
      116,
      114,
      117,
      101,
      44,
      102,
      97,
      108,
      115,
      101,
      10,
    ],
    "fileBase": "c",
    "fileExt": ".csv",
    "fileName": "c.csv",
    "path": "/tmp/foo/c.csv",
    "relDir": ".",
    "relPath": "c.csv",
  },
  {
    "_id": "c367b94579a7a9ac870a1b33a35075533487d8d6",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      123,
      34,
      98,
      97,
      109,
      34,
      58,
      32,
      116,
      114,
      117,
      101,
      44,
      32,
      34,
      98,
      97,
      122,
      34,
      58,
      32,
      32,
      102,
      97,
      108,
      115,
      101,
      32,
      125,
    ],
    "fileBase": "d",
    "fileExt": ".json",
    "fileName": "d.json",
    "path": "/tmp/foo/d.json",
    "relDir": ".",
    "relPath": "d.json",
  },
  {
    "_id": "a5e1be9ba6cb5bf18597ee658159266a000426e6",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      73,
      71,
      78,
      79,
      82,
      69,
      32,
      49,
    ],
    "fileBase": "ignore1",
    "fileExt": ".txt",
    "fileName": "ignore1.txt",
    "path": "/tmp/foo/ignore1.txt",
    "relDir": ".",
    "relPath": "ignore1.txt",
  },
  {
    "_id": "460eab01f1a4de403aced612ab4230ee610d6c09",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      69,
    ],
    "fileBase": "e",
    "fileExt": ".txt",
    "fileName": "e.txt",
    "path": "/tmp/foo/bar/e.txt",
    "relDir": "bar",
    "relPath": "bar/e.txt",
  },
  {
    "_id": "f8cba7f6ca545e7d307a508d628ef61dc9ac8072",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      70,
    ],
    "fileBase": "f",
    "fileExt": ".log",
    "fileName": "f.log",
    "path": "/tmp/foo/bar/f.log",
    "relDir": "bar",
    "relPath": "bar/f.log",
  },
  {
    "_id": "5da00cfdc5bee8aa488d13af769799a7b4043766",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "# Home
home content",
    "date": 2023-12-31T00:00:00.000Z,
    "fileBase": "index",
    "fileExt": ".md",
    "fileName": "index.md",
    "frontMatter": {
      "date": 2023-12-31T00:00:00.000Z,
      "layout": "layout.hbs",
      "title": "Home",
    },
    "layout": "layout.hbs",
    "path": "/tmp/foo/posts/index.md",
    "relDir": "posts",
    "relPath": "posts/index.md",
    "title": "Home",
  },
  {
    "_id": "af1250e6e96150b7d36c7b0448d0a11275cddb93",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "# P1
p1 content",
    "date": 2024-01-01T00:00:00.000Z,
    "draft": false,
    "fileBase": "p1",
    "fileExt": ".md",
    "fileName": "p1.md",
    "frontMatter": {
      "date": 2024-01-01T00:00:00.000Z,
      "draft": false,
      "layout": "post.hbs",
      "title": "P1",
    },
    "layout": "post.hbs",
    "path": "/tmp/foo/posts/p1.md",
    "relDir": "posts",
    "relPath": "posts/p1.md",
    "title": "P1",
  },
  {
    "_id": "b5867102be1aca22fcd0ca30e236486c4e409b13",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "# P2
p2 content",
    "date": 2024-01-02T00:00:00.000Z,
    "draft": true,
    "fileBase": "p2",
    "fileExt": ".md",
    "fileName": "p2.md",
    "frontMatter": {
      "date": 2024-01-02T00:00:00.000Z,
      "draft": true,
      "layout": "post.hbs",
      "title": "P2",
    },
    "layout": "post.hbs",
    "path": "/tmp/foo/posts/p2.md",
    "relDir": "posts",
    "relPath": "posts/p2.md",
    "title": "P2",
  },
]
`;
